qiime/core/result_base.py

```diff
@@ -26,6 +28,14 @@ def load(cls, filepath):
         result._assert_valid_type(result.type)
         return result
 
+    @classmethod
+    def extract(cls, filepath, output_dir=None):
```

Comment:
I see now that `extract` is different from `Archiver._extract_data`; one extracts the entire archive while the other only extracts the `data` directory. I thought we'd be able to nix `_extract_data` in favor of this method, which was why I wanted `output_dir` to default to a temporary directory. Since we aren't using `extract` internally, I think `output_dir` should become a required parameter (unless there was another reason to keep the current default behavior).